src/backend/distributed/citus--7.4-1--7.4-2.sql

```diff
@@ -0,0 +1,14 @@
+/* citus--7.4-1--7.4-2 */
+SET search_path = 'pg_catalog';
+
+DROP FUNCTION IF EXISTS worker_hash_partition_table(bigint, integer, text, text, oid, integer);
```

Comment:
This function might actually get used right now, which means users cannot upgrade their workers without upgrading the master at the same time, which is a common pattern. Is there a way to keep this function?

Reply:
Well, it turned out that we cannot do that without code duplication.

The reason is that Citus 7.3's hash pruning is different that Citus 7.4's hash pruning in the `worker_hash_partition_table()`. So, some workers relying on Citus shard pruning and the others relying on old shard pruning implemented in `worker_hash_partition_table()` ends up wrong query results.

So, if we really want to do that, we'd end up not removing any code from the older implementation and add the new implementation. We'd also probably end-up with some not nice `if` statements, acting according to the version.

How important to not break users during the upgrades? Shall we duplicate the code or fail hash repartitioning queries during the upgrades?